delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutesSpec.scala

```diff
@@ -534,11 +538,73 @@ class ProjectsRoutesSpec extends BaseRouteSpec with IOFromMap {
             orgUuid,
             rev = 4,
             deprecated = true,
-            markedForDeletion = true
+            markedForDeletion = true,
+            createdBy = userWithCreatePermission,
+            updatedBy = userWithDeletePermission
           )
         )
       }
     }
+
+    "fail to undeprecate a project without projects/write permission" in {
+      val (org, project) = thereIsADeprecatedProject
+      Put(s"/v1/projects/$org/$project/undeprecate?rev=2") ~> as(userWithReadPermission) ~> routes ~> check {
+        response.shouldBeForbidden
+      }
+    }
+
+    "fail to undeprecate a project if it is not deprecated" in {
+      val (org, project) = thereIsAProject
+      Put(s"/v1/projects/$org/$project/undeprecate?rev=1") ~> as(superUser) ~> routes ~> check {
+        status shouldEqual StatusCodes.BadRequest
+        response.asJson shouldEqual jsonContentOf(
+          "projects/errors/project-not-deprecated.json",
+          "org"  -> org,
+          "proj" -> project
+        )
+      }
+    }
```

Comment:
Is it fair to say this does not check that the project is actually still active? It should also be possible to test this in `ProjectsImplSpec` instead and leave this as an error serialization test only

Reply:
Good spot :)

I was trying to avoid changing `ProjectsImplSpec` though as I was anticipating the same issues I had with this one. Do you think it's worth the investment?

Reply:
I don't think it needs to be done here. But in my experience the ImplSpec were easier to work with than the Route ones

Reply:
Maybe something in the integration tests ?

Reply:
I purposely didn't add edge cases to the integration tests as they are so much slower, and now really better than testing at this level. The integration test has the "happy" path only, which seems like the correct way to test given that these route specs are almost end-to-end tests